cmd/kube-controller-manager/app/plugins_providers.go

```diff
@@ -28,11 +32,21 @@ import (
 	"k8s.io/kubernetes/pkg/volume/vsphere_volume"
 )
 
+type probeFn func() []volume.VolumePlugin
+
+func probePlugin(plugins []volume.VolumePlugin, key featuregate.Feature, fn probeFn) []volume.VolumePlugin {
+	if feature.DefaultFeatureGate.Enabled(features.CSIMigrationInTreeOff) && feature.DefaultFeatureGate.Enabled(key) {
```

Comment:
the reason for doing this isn't really self-documenting. Could you add a comment here (and at the other `probePlugin` function) explaining *why* we are removing this plugin (or link to the lines in the design)